src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -75,6 +79,97 @@ public NetworkModificationService(NetworkStoreService networkStoreService, Netwo
         return Flux.fromStream(() -> modificationRepository.getModifications(groupUuid).stream());
     }
 
+    private boolean disconnectLineBothSides(Network network, String lineId) {
+        Terminal terminal1 = network.getLine(lineId).getTerminal1();
+        boolean terminal1Disconnected = !terminal1.isConnected();
+        if (!terminal1Disconnected) {
+            terminal1Disconnected = terminal1.disconnect();
+        }
+        Terminal terminal2 = network.getLine(lineId).getTerminal2();
+        boolean terminal2Disconnected = !terminal2.isConnected();
+        if (!terminal2Disconnected) {
+            terminal2Disconnected = terminal2.disconnect();
+        }
+        return terminal1Disconnected && terminal2Disconnected;
+    }
+
+    public Flux<ElementaryModificationInfos> lockoutLine(UUID networkUuid, String lineId) {
+        return getNetwork(networkUuid)
+                .filter(network -> network.getLine(lineId) != null)
+                .switchIfEmpty(Mono.error(new NetworkModificationException(LINE_NOT_FOUND, lineId)))
+                .flatMapIterable(network -> doModification(network, networkUuid, () -> {
+                    if (disconnectLineBothSides(network, lineId)) {
+                        network.getLine(lineId).newExtension(BranchStatusAdder.class).withStatus(BranchStatus.Status.PLANNED_OUTAGE).add();
+                    } else {
+                        throw new NetworkModificationException(MODIFICATION_ERROR, "Unable to disconnect both line ends");
```

Comment:
Unit test with MODIFICATION_ERROR ?

Reply:
Need to do undo ?

Reply:
Unit tests added. The undo problem will be treated in another PR.